src/coreComponents/linearAlgebra/interfaces/direct/SuperLU_Dist.cpp

```diff
@@ -26,11 +26,13 @@ namespace geosx
 
 // Check matching requirements on index/value types between GEOSX and SuperLU_Dist
 
+#if !defined(GEOSX_USE_HYPRE_CUDA)
```

Comment:
we cannot configure SuperLU index types...so we have to turn it off when when we are using `int` for our index types.

Reply:
I guess my question was, if things work (presumably) when they're different sizes, do we need it?

Reply:
It won't compile much less work on lassen with `localIndex` being an `int`. It would be bad to call supreLU on lassen when `GEOSX_USE_HYPRE_CUDA` is enabled.

Reply:
Okay cool. But GEOSX still builds with `SuperLU_Dist` then? All the calls are just `#ifdef`d out?

Reply:
> we cannot configure SuperLU index types...

Actually we should be able to (at least as of latest release), unless someone has tried and it didn't work?
https://github.com/xiaoyeli/superlu_dist/blob/9b05b5e09fb950682cd9c367fad5fc465a0efa1a/SRC/superlu_defs.h#L81-L93
```c
/* Define my integer size int_t */
#ifdef _CRAY
  typedef short int_t;
  /*#undef int   Revert back to int of default size. */
  #define mpi_int_t   MPI_SHORT
#elif defined (_LONGINT)
  typedef int64_t int_t;
  #define mpi_int_t   MPI_LONG_LONG_INT
  #define IFMT "%ld"
#else /* Default */
  typedef int int_t;
  #define mpi_int_t   MPI_INT
  #define IFMT "%8d"
#endif
```

Reply:
That seems to be the flaw in our plan....the interoperability between the TPL's themselves.